swap_meet/vendor.py

```diff
@@ -1,2 +1,107 @@
+        '''
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        self.add(other_vendor.remove(their_item))
+        other_vendor.add(self.remove(my_item))
+        return True
+
+    def swap_first_item(self, other_vendor):
+        '''
+        Swaps the first items of two lists of inventories.
+        Inputs: other_vendor (an instance of Vendor)
+        Output: Returns True after swapping the first items in two lists of vendor inventories. False if either itself or other_vendor has an empty inventory.
+        '''
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        
+        self.add(other_vendor.remove(other_vendor.inventory[0]))
+        other_vendor.add(self.remove(self.inventory[0]))
+        return True
+
+    def get_by_category(self, category):
+        '''
+        Returns a list of item from inventory in a specific category
+        Input : category of the item (string)
+        Outputs: list of item in a specific category, Returns empty list if there is no item in the category.
+        '''
+        category_list = []
+        for item in self.inventory:
+            if item.get_category() == category:
+                category_list.append(item)
+
+        return category_list       
+                
+    def get_best_by_category(self, category):
+        '''
+        Returns a best item in a category
+        Input : category of the item (string)
+        Outputs: return item with best category, Returns None if there is no item in the category.
+        '''
+        highest_condition = 0
+        best_category_item = None
+        for item in self.inventory:
+            if item.get_category() == category and item.condition > highest_condition:
+                highest_condition = item.condition
+                best_category_item = item
+        return best_category_item        
+
+    def swap_best_by_category(self, other_vendor, my_priority, their_priority):
+        '''
+        Finds and swaps the best item of a certain category with another vendor.
+        Inputs: other_vendor (an instance of Vendor), my_priority (a string representing the item category the user wants swapped into their inventory), their_priority (a string representing the item category the other vendor wants swapped into their inventory)
+        Output: Returns True is swap was possible, False if Vendor has no item that matches their_priority category or other_vendor has no item that matches my_priority category.
+        '''
+        if (self.get_best_by_category(their_priority) is None or 
+            other_vendor.get_best_by_category(my_priority) is None):
+            return False
+        
+        vendor_receives = self.get_best_by_category(their_priority)
+        other_vendor_receives = other_vendor.get_best_by_category(my_priority)
+
+        self.swap_items(other_vendor, vendor_receives, other_vendor_receives)
+        return True
```

Comment:
If you initialize `vendor_receives` and `other_vendor_receives` first, you won't have to type `self.get_best_by_category(their_priority)` and `other_vendor.get_best_by_category(my_priority)` multiple times.

```suggestion
        vendor_receives = self.get_best_by_category(their_priority)
        other_vendor_receives = other_vendor.get_best_by_category(my_priority)

        if vendor_receives is None or other_vendor_receives is None:
            return False
        
        return self.swap_items(
            other_vendor, 
            vendor_receives, 
            other_vendor_receives
        )
```